nilearn/datasets/atlas.py

```diff
@@ -963,3 +966,149 @@ def fetch_atlas_surf_destrieux(data_dir=None, url=None,
+        for region_nb, region in enumerate(labels[:, pos]):
+            level_labels.setdefault(region, len(level_labels))
+            level_img[atlas_img.get_data() == region_nb] = level_labels[
+                region]
+        level_img <<= 8 * pos
+        new_img |= level_img
+        level_labels = list(list(
+            zip(*sorted(level_labels.items(), key=lambda t: t[1])))[0])
+        level_labels[0] = 'Background'
+        levels.append((level, level_labels))
+    new_img = new_img_like(atlas_img, data=new_img)
+    return new_img, levels
+
+
+def _get_talairach_all_levels(data_dir=None, verbose=1):
+    """Get the path to Talairach atlas and labels
+
+    The atlas is downloaded and the files are created if necessary.
+
+    The image contains all five levels of the atlas, each encoded on 8 bits
+    (least significant octet contains the hemisphere, the next one the lobe,
+    then gyrus, tissue, and ba).
+
+    The labels json file contains
+    [['level name', ['labels', 'for', 'this', 'level' ...]], ...],
+    where the levels are in the order mentionned above.
+
+    """
+    data_dir = _get_dataset_dir('talairach_atlas', data_dir=data_dir)
+    img_file = os.path.join(data_dir, 'talairach.nii')
+    labels_file = os.path.join(data_dir, 'talairach_labels.json')
+    if os.path.isfile(img_file) and os.path.isfile(labels_file):
+        return img_file, labels_file
+    atlas_url = 'http://www.talairach.org/talairach.nii'
+    temp_dir = mkdtemp()
+    try:
+        temp_file = _fetch_files(
+            temp_dir, [('talairach.nii', atlas_url, {})], verbose=verbose)[0]
+        atlas_img = nb.load(temp_file, mmap=False)
+        atlas_img = check_niimg(atlas_img)
+    finally:
+        shutil.rmtree(temp_dir)
+    labels = atlas_img.header.extensions[0].get_content()
+    labels = labels.strip().decode('utf-8').split('\n')
+    labels = [l.split('.') for l in labels]
+    new_img, level_labels = _separate_talairach_levels(
+        atlas_img, labels, verbose=verbose)
+    new_img.to_filename(img_file)
+    with open(labels_file, 'w') as fp:
+        json.dump(level_labels, fp)
+    return img_file, labels_file
+
+
+def fetch_atlas_talairach(level_name, data_dir=None, verbose=1):
+    """Download the Talairach atlas.
+
+    .. versionadded:: 0.3.2
+
+    Parameters
+    ----------
+    level_name : {'hemisphere', 'lobe', 'gyrus', 'tissue', 'ba'}
+        Which level of the atlas to use: the hemisphere, the lobe, the gyrus,
+          the tissue type or the Brodmann area.
+
+    data_dir : str, optional (default=None)
+        Path of the data directory. Used to force data storage in a specified
+        location.
+
+    verbose : int
+        verbosity level (0 means no message).
+
+    Returns
+    -------
+    sklearn.datasets.base.Bunch
+        Dictionary-like object, contains:
+
+        - maps: 3D Nifti image, values are indices in the list of labels.
+        - labels: list of strings. Starts with 'Background'.
```

Comment:
Small question: Are these labels sorted according to maps indices ?

Reply:
yes, should I repeat this here or is 'values are indices in the list of labels' enough?

Reply:
I think that should be enough.